Artesian/Artesian.SDK/Service/Query/VersionedQuery.cs

```diff
@@ -270,7 +274,15 @@ public VersionedQuery ForVersion(LocalDateTime version)
         /// <returns>Enumerable of TimeSerieRow Versioned</returns>
         public async Task<IEnumerable<TimeSerieRow.Versioned>> ExecuteAsync(CancellationToken ctk = default)
         {
-            return await _client.Exec<IEnumerable<TimeSerieRow.Versioned>>(HttpMethod.Get, _buildRequest(), ctk: ctk);
+            List<string> urls = _buildRequest();
+
+            var taskList = urls.Select(url => _client.Exec<IEnumerable<TimeSerieRow.Versioned>>(HttpMethod.Get, url, ctk: ctk));
+
+            await Task.WhenAll(taskList.ToArray());
+
+            var res = taskList.SelectMany(t => t.GetAwaiter().GetResult());
```

Comment:
ELIMINATE any `GetResult()` from any async flow. Use await.
And btw WhenAll() returns what you need already, just use that.

If you ever use `GetResult()` again in an Async method, remember that for sure is wrong.